src/Adapter/LegacyContext.php

```diff
@@ -100,7 +102,15 @@ public function getAdminBaseUrl()
      */
     public function getAdminLink($controller, $withToken = true, $extraParams = array())
     {
-        return $this->getContext()->link->getAdminLink($controller, $withToken, $extraParams, $extraParams);
+        $id_lang = Context::getContext()->language->id;
+        $params = $extraParams;
+        if ($withToken) {
+            $params['token'] = ToolsLegacy::getAdminTokenLite($controller);
+        }
+
+        $link = new Link();
+
+        return $link->getAdminBaseLink() . basename(_PS_ADMIN_DIR_) . '/' . Dispatcher::getInstance()->createUrl($controller, $id_lang, $params, false);
```

Comment:
This should not be modified as global fix on migrated urls is coming soon, better not multiply the places where we generate the urls